mezzanine/generic/templatetags/comment_tags.py

```diff
@@ -23,11 +23,11 @@ def comments_for(context, obj):
     """
     form_class = import_dotted_path(settings.COMMENT_FORM_CLASS)
     form = form_class(context["request"], obj)
-    try:
-        context["posted_comment_form"]
-    except KeyError:
+    if context.get("posted_comment_form") is None or \
+                    context.get("posted_comment_form") != form:
```

Comment:
I'm pretty sure this will always evaluate `True`. That is, the form we just instantiated locally will never be equal to the form instantiated in the `comment` view and passed into the context.

A more useful check might be to compare `form.target_object`?

```python

context_form = context.get("posted_comment_form")

return {
    'posted_comment_form': context_form if context_form.target_object == obj else form,
    'unposted_comment_form': form,
    'comment_url': reverse("comment"),
    'object_for_comments': obj,
    'request': context.get('request'),
}
```

I'm returning a new context here to avoid introducing context variables into the global context that are only relevant to `generic/includes/comments.html`.